### DISCUSSION TITLE
(node:6572) [DEP0005] DeprecationWarning: Buffer() is deprecated due to security and usability issues. Please use the Buffer.alloc(), Buffer.allocUnsafe(), or Buffer.from() methods instead. (Use `node --trace-deprecation ...` to show where the warning was created)

### DISCUSSION THREAD

**Discussion by shriekdj** [vercel/next.js#58170] [Help] [answered]
### Summary

 I am getting this issue for running nextjs under iis.

most of the time the nextjs app works fine there with nextjs's custom server with expressjs under iis.

this issue only happenes on live production server.

```js
(node:6572) [DEP0005] DeprecationWarning: Buffer() is deprecated due to security and usability issues. Please use the Buffer.alloc(), Buffer.allocUnsafe(), or Buffer.from() methods instead.
(Use `node --trace-deprecation ...` to show where the warning was created)
```

### Additional information

```js
App Deployed on : IIS
NextJS Version : 13.5.4
Using App Router : true
Using Custom Server : true
Custom Server Program : express.js boilerplate.
Occurrence of error : undefined

issue only happened on live server, suddenly, mostly occures once a day and we had to restart the server and issue solves but we just can't live with it like that.
```

**Comment by icyJoseph** [comment]
What's your Node version?

**Reply by shriekdj** [reply]
i am using lts version v20.9.0

**Comment by mutasim77** [comment]
Update both nodejs and npm to the latest versions. After that go to your project and delete the **node_modules** folder and re-install it by ```npm install```

**Reply by shriekdj** [reply]
i already tried that, but this issue only occures on production server

**Reply by Saprou** [reply]
same here, did you fix it?

**Reply by shriekdj** [reply, marked as answer]
@Saprou my issue solved when i updated my tsconfig file to use `ES2015` and above.
I Am Now using `ES2020`

**Reply by vstar001** [reply]
hi can you tell me that 
 answer iam using javascript

**Comment by shrekuu** [comment]
I have the same issue. It's not related to Next.js. I guess you are using web3.js 1.x.  It's from this package. You can just ignore it,  or be brave and upgrade to the latest web3.js v4. It might take a lot of refactoring work. 
https://stackoverflow.com/questions/52165333/deprecationwarning-buffer-is-deprecated-due-to-security-and-usability-issues

**Comment by shriekdj** [comment]
@shrekuu  my issue solved when i updated my tsconfig file to use ES2015 and above.
I Am Now using ES2020

**Reply by UTSProjects** [reply]
But i am not using typescript , so i dont have a tsconfig file , what should i do ?
(node:10116) [DEP0005] DeprecationWarning: Buffer() is deprecated due to security and usability issues. Please use the Buffer.alloc(), Buffer.allocUnsafe(), or Buffer.from() methods instead.
(Use `node --trace-deprecation ...` to show where the warning was created)

**Reply by shriekdj** [reply]
if you are not using typescript, then you can use babel.config.js instead for your javascript program and set it's target to ES2015 and above. everything else is same. you will not gonna need to change your project to typescript.

**Reply by UTSProjects** [reply]
For some reason that was not the issue that was causing it ,IG there was some issue with ports : I was using standalone feature of nextjs and in there my entrypoint is server.js file which had 
const currentPort = parseInt(process.env.PORT || 3000) I changed it to 
const currentPort = process.env.PORT || 3000 and it worked for me.

**Reply by shriekdj** [reply]
@UTSProjects cool. your issue solved that is what important, then whatever method 👍 .
